File: src/JavaScript/medium/array/problem2433.js
/**
 * @param {number[]} pref
 * @return {number[]}
 */
var findArray = function(pref) {
    let arr = []
    for(let i = 0; i < pref.length; i++){
        if(arr.length){
            arr[i] = pref[i] ^ pref[i - 1]
        } else {
            arr[0] = pref[0]
        }
    }
    return console.log(arr);
};

let pref = [5,2,0,3,1]
// let pref = [13]

findArray(pref)

File: src/Python/medium/array/problem2433.py
from typing import List

class Solution:
    def findArray(self, pref: List[int]) -> List[int]:
        arr = [0] * len(pref)
        for i in range(len(pref)):
            if i == 0:
                arr[0] = pref[0]
            else:
                arr[i] = pref[i - 1] ^ pref[i]
        return arr


# pref = [5,2,0,3,1]
pref = [13]

solution = Solution()
solution.findArray(pref)